orttraining/orttraining/training_ops/cpu/torch/torch_custom_function_kernel_base.h

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#ifndef SHARED_PROVIDER
+#include "core/common/common.h"
+#include "core/framework/op_kernel.h"
+#endif
+
+namespace onnxruntime {
+namespace contrib {
+
+std::vector<OrtValue> CreateOrtValueArgs(OpKernelContext* context,
+                                         const size_t begin_index,
+                                         const size_t num_arg);
+
+class PythonOpBase {
+ protected:
+  PythonOpBase(const OpKernelInfo& info);
+
+  void AddIntScalarArgs();
+  void AddInputTupleArgs();
+  void AddFloatTupleArgs();
+  void AddPointerScalarArgs();
+  void CreateConstArgs();
+  void CreateArgPositions();
+
+  void SetContextOutput(OpKernelContext* context, void* diff_ctx) const;
+  void SetOtherOutputs(OpKernelContext* context, std::vector<OrtValue>& returned_args) const;
+
+  std::vector<int64_t> const_arg_positions_;
+  std::vector<void*> const_args_;
+  std::vector<int64_t> arg_positions_;
+
+  // Name of containing class. For example, MyReLU.
+  std::string name_;
```

Comment:
const?

Reply:
Probably no. This field is filled by
```
  ORT_THROW_IF_ERROR(info.GetAttr("name", &name_));
```
so we can't directly do
```
  name_(info.GetAttr("name", &name_))
```
We can wrap the call `info.GetAttr("name", &name_)` as a one line function returning `std::string`, but I feel it decreases readability.